src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -566,6 +566,18 @@ impl AccessChecker for PrivilegeAccess {
     #[async_backtrace::framed]
     async fn check(&self, ctx: &Arc<QueryContext>, plan: &Plan) -> Result<()> {
         let user = self.ctx.get_current_user()?;
+        // if user need change password, only change self password is allowed.
+        let need_change = user.auth_info.get_need_change();
+        if need_change {
+            if let Plan::AlterUser(plan) = plan {
+                if plan.user.username == user.name && plan.user_option.is_none() {
+                    return Ok(());
+                }
+            }
+            return Err(ErrorCode::PermissionDenied(
+                "Must change password before execute other operations".to_string(),
```

Comment:
If MUST_CHANGE_PASSWORD=true is set, also support login but need to modify password first?

Reply:
Maybe use other error code?

Reply:
add a new error code `NeedChangePasswordDenied`